### ISSUE TITLE
Fix PageTransition unpleasant flickering glitch (solution provided)

### ISSUE THREAD

**Issue by dbriard** [AvaloniaUI/Avalonia#11476]
**Is your feature request related to a problem? Please describe.**
I've noticed that provided PageTransitions such as PageSlide have a flickering glitch.
When the transition starts, at the very begining, the "to" visual is displayed over the "from" visual during 1 frame, which produce a unpleasant flickering effect.

I think this is due to the IsVisible=true that happen before the animation starts ( when the translation is not yet set to parent width).
```
if (to != null)
            {
                to.IsVisible = true;
                var animation = new Animation
                {
```

**Describe the solution you'd like**
I would like the flickering glitch is fixed for all provided transitions.

**Describe alternatives you've considered**
An easy fix is to set Opacity to zero at the same time that IsVisible is set to true. So during the 1 frame that the "to" visual is visible at the wrong place, it will be hidden by zero opacity.

I modified PageSlide animation to fix the glitch like that (see the added lines).

```
var opacityProperty = Visual.OpacityProperty; // added
            
            if (to != null)
            {
                to.Opacity = 0; // added
                to.IsVisible = true;
                var animation = new Animation
                {
                    Easing = SlideInEasing,
                    Children =
                    {
                        new KeyFrame
                        {
                            Setters =
                            {
                                new Setter(translateProperty, forward ? distance : -distance),
                                new Setter(opacityProperty, 1d), // added
                            },
                            Cue = new Cue(0d)
                        },
                        new KeyFrame
                        {
                            Setters = 
                            { 
                                new Setter(translateProperty, 0d),
                                new Setter(opacityProperty, 1d) // added
                            },
                            Cue = new Cue(1d)
                        }
                    },
                    Duration = Duration
                };
                tasks.Add(animation.RunAsync(to, null, cancellationToken));
            }

            await Task.WhenAll(tasks);

            if (!cancellationToken.IsCancellationRequested)
            {
                if (from != null)
                {
                    from.IsVisible = false;
                }
                if (to != null) // added
                {
                    to.Opacity = 1; // added
                }
            }
```

**Comment by grokys** [comment]
Yeah there's still a problem with the one page as discussed in #11326 - @kekekeks is working on synchronizing the animation timer and the render timer which should fix this problem, hopefully we'll have a solution in a few days.

**Comment by dbriard** [comment]
> and the render timer which should fix this problem,

Ah good to know, so my workaround with opacity will not be needed anymore, it will be easier to make custom transitions. Thanks for the good work on this awesome framework!

**Comment by danielmayost** [comment]
Fixed by #11552? i can't reproduce this flicker anymore.

**Comment by maxkatz6** [comment]
Yes, it should. I also can't reproduce it anymore.
